src/lib/role-permission-keto-tuples.ts

```diff
@@ -76,45 +72,49 @@ class KetoTuples {
   async updateAllRolePermissions (rolePermissionCombos: string[]) : Promise<void> {
     const currentPermissionCombos = (await this.getAllRolePermissionCombos()) || []
     // Calculate the permissions to be deleted
-    const deletePermissionCombos = currentPermissionCombos?.filter(item => !rolePermissionCombos.includes(item))
+    const deletePermissionCombos = currentPermissionCombos?.filter(item => !rolePermissionCombos?.includes(item))
     // Calculate the permissions to be added
     const addPermissionCombos = rolePermissionCombos?.filter(item => !currentPermissionCombos.includes(item))
 
     // Construct patch delta
-    const patchDeltaArr = this._generateRolePermissionComboPatchDeltaArray(addPermissionCombos, deletePermissionCombos)
-    if (patchDeltaArr.length > 0) {
+    const relationshipPatch = this._generateRolePermissionComboPatchDeltaArray(
+      addPermissionCombos, deletePermissionCombos
+    )
+    if (relationshipPatch.length > 0) {
       logger.info('Patching permissions in Ory Keto....')
-      await this.oryKetoWriteApi.patchRelationTuples(patchDeltaArr)
+      await this.adminRelationshipApi.patchRelationships({ relationshipPatch })
     }
   }
 
   _generateRolePermissionComboPatchDeltaArray (
     addPermissionCombos: string[], deletePermissionCombos: string[]
-  ): PatchDelta[] {
-    let patchDeltaArray: PatchDelta[] = []
+  ): RelationshipPatch[] {
+    let patchDeltaArray: RelationshipPatch[] = []
+
     patchDeltaArray = patchDeltaArray.concat(addPermissionCombos.map(permissionCombo => {
       const rolePermissionArr = permissionCombo.split(':')
       return {
-        action: 'insert',
+        action: RelationshipPatchActionEnum.Insert,
         relation_tuple: this._generatePermissionTuple(rolePermissionArr[0], rolePermissionArr[1])
       }
     }))
     patchDeltaArray = patchDeltaArray.concat(deletePermissionCombos.map(permissionCombo => {
       const rolePermissionArr = permissionCombo.split(':')
       return {
-        action: 'delete',
+        action: RelationshipPatchActionEnum.Delete,
         relation_tuple: this._generatePermissionTuple(rolePermissionArr[0], rolePermissionArr[1])
       }
     }))
+
     return patchDeltaArray
   }
 
-  _generatePermissionTuple (role: string, permission: string): InternalRelationTuple {
+  _generatePermissionTuple (role: string, permission: string): Relationship {
     return {
-      namespace: 'permission',
+      namespace: KETO_NAMESPACES.permission,
       object: permission,
-      relation: 'granted',
-      subject: 'role:' + role + '#member'
+      relation: KETO_RELATIONS.granted,
+      subject_id: `role:${role}#member`
```

Comment:
Is it working. I think we can use subject_set for this.
Instead of `role:ROLEID#member`, we can use the following
```
    "subject_set": {
        "namespace": "role",
        "object": "ROLEID",
        "relation": "member"
    }
```

Reply:
fixed